pkg/apiserver/api_installer.go

```diff
@@ -216,18 +216,27 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	}
 	versionedStatus := indirectArbitraryPointer(versionedStatusPtr)
 	var (
-		getOptions          runtime.Object
-		versionedGetOptions runtime.Object
-		getOptionsKind      string
-		getSubpath          bool
-		getSubpathKey       string
+		getOptions               runtime.Object
+		versionedGetOptions      runtime.Object
+		getOptionsInternalFQKind unversioned.GroupVersionKind
+		getOptionsExternalFQKind unversioned.GroupVersionKind
+		getSubpath               bool
+		getSubpathKey            string
 	)
 	if isGetterWithOptions {
 		getOptions, getSubpath, getSubpathKey = getterWithOptions.NewGetOptions()
-		_, getOptionsKind, err = a.group.Typer.ObjectVersionAndKind(getOptions)
+		getOptionsGVString, getOptionsKind, err := a.group.Typer.ObjectVersionAndKind(getOptions)
 		if err != nil {
 			return nil, err
 		}
+		gv, err := unversioned.ParseGroupVersion(getOptionsGVString)
+		if err != nil {
+			return nil, err
+		}
+		getOptionsInternalFQKind = gv.WithKind(getOptionsKind)
```

Comment:
Is this just a temporary arrangement? ObjectVersionAndKind will return a GroupVersionWithKind eventually, right?